ecs-agent/netlib/common_test.go

```diff
@@ -21,3 +46,215 @@ const (
 	searchDomainName   = "us-west-2.test.compute.internal"
 	netNSPathDir       = "/var/run/netns/"
 )
+
+// getSingleNetNSAWSVPCTestData returns a task payload and a task network config
+// to be used the input and reference result for tests. The reference object will
+// has only one network namespace and network interface.
+func getSingleNetNSAWSVPCTestData(testTaskID string) (*ecsacs.Task, tasknetworkconfig.TaskNetworkConfig) {
+	enis, netIfs := getTestInterfacesData()
+	taskPayload := &ecsacs.Task{
+		NetworkMode:              aws.String(ecs.NetworkModeAwsvpc),
+		ElasticNetworkInterfaces: []*ecsacs.ElasticNetworkInterface{enis[0]},
+	}
+
+	netNSName := fmt.Sprintf(netNSNamePattern, testTaskID, eniName)
+	netNSPath := netNSPathDir + netNSName
+	taskNetConfig := tasknetworkconfig.TaskNetworkConfig{
+		NetworkMode: ecs.NetworkModeAwsvpc,
+		NetworkNamespaces: []*tasknetworkconfig.NetworkNamespace{
+			{
+				Name:  netNSName,
+				Path:  netNSPath,
+				Index: 0,
+				NetworkInterfaces: []*networkinterface.NetworkInterface{
+					&netIfs[0],
+				},
+				KnownState:   status.NetworkNone,
+				DesiredState: status.NetworkReadyPull,
+			},
+		},
+	}
+
+	return taskPayload, taskNetConfig
+}
+
+// getSingleNetNSMultiIfaceAWSVPCTestData returns test data for EKS like use cases.
+func getSingleNetNSMultiIfaceAWSVPCTestData(testTaskID string) (*ecsacs.Task, tasknetworkconfig.TaskNetworkConfig) {
+	taskPayload, taskNetConfig := getSingleNetNSAWSVPCTestData(testTaskID)
+	enis, netIfs := getTestInterfacesData()
+	secondIFPayload := enis[1]
```

Comment:
Can we have some len check here? Or have some unit tests to help us ensure we won't run into nil pointer error?